lockable/lockable.py

```diff
@@ -63,68 +64,18 @@ def __init__(self, hostname=socket.gethostname(),
         self._lock_folder = lock_folder
-        self._resource_list = None
-        self._resource_list_file_mtime = None
-        self._resource_list_file = resource_list_file
         assert not (isinstance(resource_list, list) and
                     resource_list_file), 'only one of resource_list or ' \
                                          'resource_list_file is accepted, not both'
-        if isinstance(self._resource_list_file, str):
-            self._resource_list_file_mtime = os.path.getmtime(resource_list_file)
-            self.load_resources_list_file(self._resource_list_file)
-        elif isinstance(resource_list, list):
-            self.load_resources_list(resource_list)
+        if resource_list is None and resource_list_file is None:
+            self._provider = create_provider([])
         else:
-            self.logger.warning('resource_list_file or resource_list is not configured')
+            self._provider = create_provider(resource_list_file or resource_list)
 
-    def load_resources_list_file(self, filename: str):
-        """ Load resources list file"""
-        self.load_resources_list(self._read_resources_list(filename))
-        self.logger.warning('Use resources from %s file', filename)
-
-    def load_resources_list(self, resources_list: list):
-        """ Load resources list """
-        assert isinstance(resources_list, list), 'resources_list is not an list'
-        self._resource_list = resources_list
-        self.logger.debug('Resources loaded: ')
-        for resource in self._resource_list:
-            self.logger.debug(json.dumps(resource))
-
-    def reload_resource_list_file(self):
-        """ Reload resources from file if file has been modified """
-        if self._resource_list_file_mtime is None:
-            return
-
-        mtime = os.path.getmtime(self._resource_list_file)
-        if self._resource_list_file_mtime != mtime:
-            self._resource_list_file_mtime = mtime
-            self.load_resources_list_file(self._resource_list_file)
-
-    @staticmethod
-    def _read_resources_list(filename):
-        """ Read resources json file """
-        MODULE_LOGGER.debug('Read resource list file: %s', filename)
-        with open(filename) as json_file:
-            try:
-                data = json.load(json_file)
-                assert isinstance(data, list), 'data is not an list'
-            except (json.decoder.JSONDecodeError, AssertionError) as error:
-                raise ValueError(f'invalid resources json file: {error}') from error
-            Lockable._validate_json(data)
-        return data
-
-    @staticmethod
-    def _validate_json(data):
-        """ Internal method to validate resources.json content """
-        counts = count_by(data, lambda obj: obj.get('id'))
-        no_ids = filter_(counts.keys(), lambda key: key is None)
-        if no_ids:
-            raise ValueError('Invalid json, id property is missing')
+    @property
+    def _resource_list(self):
```

Comment:
Calling this triggers http call each time when http provider is used.

Reply:
fixed in 6f519e9